File: app/src/main/java/com/iceteaviet/fastfoodfinder/data/transport/model/SearchEventResult.java
package com.iceteaviet.fastfoodfinder.data.transport.model;

/**
 * Created by Genius Doan on 11/28/2016.
 */
public class SearchEventResult {
    public static final int SEARCH_ACTION_QUICK = 0;
    public static final int SEARCH_ACTION_QUERY_SUBMIT = 1;
    public static final int SEARCH_ACTION_COLLAPSE = 2;
    private int mResult;
    private String mSearchString;
    private int mStoreType;

    public SearchEventResult(int resultCode, String searchString, int storeType) {
        mResult = resultCode;
        mSearchString = searchString;
        mStoreType = storeType;
    }

    public SearchEventResult(int resultCode, int storeType) {
        mResult = resultCode;
        mStoreType = storeType;
    }

    public SearchEventResult(int resultCode, String searchString) {
        mResult = resultCode;
        mSearchString = searchString;
    }

    public SearchEventResult(int resultCode) {
        mResult = resultCode;
    }

    public int getResultCode() {
        return mResult;
    }

    public String getSearchString() {
        return mSearchString;
    }

    public int getStoreType() {
        return mStoreType;
    }
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/data/transport/model/SearchEventResult.kt
package com.iceteaviet.fastfoodfinder.data.transport.model

/**
 * Created by Genius Doan on 11/28/2016.
 */
class SearchEventResult {
    var resultCode: Int = 0
        private set
    var searchString: String? = null
        private set
    var storeType: Int = 0
        private set

    constructor(resultCode: Int, searchString: String, storeType: Int) {
        this.resultCode = resultCode
        this.searchString = searchString
        this.storeType = storeType
    }

    constructor(resultCode: Int, storeType: Int) {
        this.resultCode = resultCode
        this.storeType = storeType
    }

    constructor(resultCode: Int, searchString: String) {
        this.resultCode = resultCode
        this.searchString = searchString
    }

    constructor(resultCode: Int) {
        this.resultCode = resultCode
    }

    companion object {
        const val SEARCH_ACTION_QUICK = 0
        const val SEARCH_ACTION_QUERY_SUBMIT = 1
        const val SEARCH_ACTION_COLLAPSE = 2
    }
}
